src/app/search/components/filter/filter-provider.component.tsx

```diff
@@ -1,38 +1,52 @@
 "use client";
 
-import React, { Dispatch, PropsWithChildren, SetStateAction } from "react";
+import React, { createContext, useReducer } from "react";
 
-import { createContext, useState } from "react";
-
-type SelectedFilters = {
+type Filters = {
   service: string;
-  specialtie: string;
-  serviceType: string;
+  expertise: string;
+  plural: string;
 };
 
+type Action =
+  | { type: "SET_SERVICE"; payload: string }
+  | { type: "SET_EXPERTISE"; payload: string }
+  | { type: "SET_PLURAL"; payload: string };
```

Comment:
وقتی همشون یه جوره، نیاز نیست جدا باشن

```suggestion
type Action = {
  type: "SET_SERVICE" | "SET_EXPERTISE" | "SET_PLURAL";
  payload: string;
};
```